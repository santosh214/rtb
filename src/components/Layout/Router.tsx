import { Navigate, Outlet, Route, Routes } from 'react-router-dom';
import AdminHome from '../../pages/Admin/AdminHome';
import Users from '../../pages/Admin/users/Users';
import Transactions from '../../pages/Admin/Transaction/Transactions';
import AddUser from '../../pages/Admin/users/AddUser';
import Login from '../../pages/common/Login';
import UpdateUser from '../../pages/Admin/users/UpdateUser';
import AdminItem from '../../pages/Admin/Items/Item';
import AddItem from '../../pages/Admin/Items/AddItems';
import UpdateItems from '../../pages/Admin/Items/UpdateItems';
import SignUp from '../../pages/common/SignUp';
import UnAuthorized from '../../pages/common/UnAuthorized';
import { getAdmin } from '../../utils/utlis';

const PrivateRoutes = () => {
  return getAdmin() ? <Outlet /> : <Navigate to="/unauthorized" />;
};

export default function Router() {
  return (
    <Routes>
      {/* Private Routes - Require authentication and admin role */}
      <Route element={<PrivateRoutes />}>
        <Route path="/dashboard" element={<AdminHome />} />
        <Route
          path="/dashboard/items"
          element={<AdminItem employee={false} />}
        />
        <Route path="/dashboard/items/add" element={<AddItem />} />
        <Route path="/dashboard/items/:id" element={<UpdateItems />} />
        <Route path="/dashboard/user/:id" element={<UpdateUser />} />
        <Route path="/dashboard/transactions" element={<Transactions />} />
        <Route path="/dashboard/users/add" element={<AddUser />} />
        <Route path="/dashboard/users" element={<Users />} />
      </Route>
      {/* Public Routes */}
      <Route path="/" element={<AdminItem employee={true} />} />
      <Route path="/auth/login" element={<Login />} />
      <Route path="/auth/register" element={<SignUp />} />
      <Route path="/unauthorized" element={<UnAuthorized />} />{' '}
      {/* Add Unauthorized page route */}
    </Routes>
  );
}
