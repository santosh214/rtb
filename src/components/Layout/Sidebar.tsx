import Drawer from '@mui/material/Drawer';
import List from '@mui/material/List';
import Divider from '@mui/material/Divider';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import InboxIcon from '@mui/icons-material/MoveToInbox';
import MailIcon from '@mui/icons-material/Mail';
import { useNavigate } from 'react-router-dom';
import { MenuList } from '../../utils/constant';

const drawerWidth = 240;

export default function Sidebar() {
    const navigate = useNavigate()
    const handleNavigation = (route: string) => {
        navigate(route)
    }
    return (

        <Drawer
            sx={{

                width: drawerWidth,
                flexShrink: 0,
                '& .MuiDrawer-paper': {
                    width: drawerWidth,
                    boxSizing: 'border-box',
                    position: 'unset'
                },
            }}
            variant="permanent"
            anchor="left"

        >
            <Divider />
            <List>
                {MenuList.map((items, index) => (
                    <ListItem key={items.id} disablePadding >
                        <ListItemButton onClick={() => handleNavigation(items.route)}>
                            <ListItemIcon>
                                {index % 2 === 0 ? <InboxIcon /> : <MailIcon />}
                            </ListItemIcon>
                            <ListItemText primary={items.name} />
                        </ListItemButton>
                    </ListItem>
                ))}
            </List>

        </Drawer>


    );
}
