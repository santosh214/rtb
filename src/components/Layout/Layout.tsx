import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';
import Navbar from './Navbar';
import Router from './Router';
import Sidebar from './Sidebar';

export default function Layout() {

    return (
        <>
            <Navbar />
            <Box sx={{ display: 'flex' }}>
                <CssBaseline />
                <Sidebar />
                <Box
                    component="main"
                    sx={{ bgcolor: 'background.default', p: 3 }}
                    width={'100%'}
                >
                    <Router />
                </Box>
            </Box>
        </>

    );
}
