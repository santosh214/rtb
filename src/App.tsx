import './App.css';
import Layout from './components/Layout/Layout';
import 'react-toastify/dist/ReactToastify.css';
import { ToastContainer } from 'react-toastify';

function App() {
  return (
    <>
      <ToastContainer />
      <Layout />
    </>
  );
}

export default App;
