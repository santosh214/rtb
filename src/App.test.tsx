import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import '@testing-library/jest-dom';
import Layout from './components/Layout/Layout';

// Mock Layout component to test App component in isolation
jest.mock('./components/Layout/Layout', () => () => <div>Mock Layout</div>);

describe('App Component', () => {
  test('renders ToastContainer and Layout', () => {
    render(<App />);


    // Check if Layout is rendered (mocked version in this case)
    expect(screen.getByText('Mock Layout')).toBeInTheDocument();
  });

  test('matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
