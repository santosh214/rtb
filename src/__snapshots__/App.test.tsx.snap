// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div>
    Mock Layout
  </div>
</DocumentFragment>
`;
